### DISCUSSION TITLE
"Local graphql datavar" functionality

### DISCUSSION THREAD

**Discussion by mrpsiho** [builderius/builderius#50] [Ideas]
At the moment datavars can be created inside template or global settings only. Since we are using dynamic data anywhere in the modules, it creates some disconnection when one exports layout and uses them on another page or even site. Example: collection module displays featured posts, everything is good on the page, however when exported and imported elsewhere it shows nothings since there is no proper dynamic data.

Solution: give a possibility to create grapqhql datavar within a module.
Local graphql datavar paired with composite module functionality would enable creating advanced dynamic modules. For instance, a composite module "Nav menu" would be ready in terms of proper structure and proper dynamic data.

Local graphql datavar functionality might have some restrictions:
1. we are talking about ONLY graphql type datavar and only ONE per module.
2. we are still thinking whether it should have an automatically generated name or custom one

Also, having datavars here and there certainly brings disorder. "Local graphql datavar" functionality is not meant to replace the current system, but serve as datavar fallback and/or example of graphql query. 
Examples: 
1. a developer imports some saved layout or composite module that contains a local datavar and that datavar actually creates dynamic data; great, it works just perfect! then the developer would transfer this local datavar into the template or global settings;
2. similar to the previous example, but the local datavar does not create any dynamic data because, for instance, query tries to reach CPT "hotels" whereas on this site there is no such CPT; then the developer checks the query, make necessary changes and also copy/paste the query into tmpl/global settings and then uses it from there.

**Comment by thisbit** [comment]
I love it
